File: src/jp/senchan/android/wasatter/task/TaskGetOAuthRequestUrl.java
/**
 *
 */
package jp.senchan.android.wasatter.task;

import jp.senchan.android.wasatter2.R;
import jp.senchan.android.wasatter.app.OAuthToken;
import twitter4j.TwitterException;
import android.os.AsyncTask;
import android.webkit.WebView;
import android.widget.Button;

/**
 * @author takuji
 * 
 */
public class TaskGetOAuthRequestUrl extends AsyncTask<Void, Void, Void> {
    private OAuthToken target;
    private Button btn;

    public TaskGetOAuthRequestUrl(OAuthToken target) {
        // TODO 自動生成されたコンストラクター・スタブ
        this.target = target;
    }

    @Override
    protected void onPreExecute() {
        // TODO 自動生成されたメソッド・スタブ
        btn = (Button) this.target.findViewById(R.id.button_set_token);
        btn.setClickable(false);
    }

    @Override
    protected Void doInBackground(Void... params) {
        // TODO 自動生成されたメソッド・スタブ
        try {
            this.target.request = this.target.twitter.getOAuthRequestToken();
        } catch (TwitterException e) {
            // TODO 自動生成された catch ブロック
            e.printStackTrace();
        }
        return null;
    }

    @Override
    protected void onPostExecute(Void result) {
        // TODO 自動生成されたメソッド・スタブ
        WebView wv = (WebView) this.target.findViewById(R.id.web);
        wv.loadUrl(this.target.request.getAuthenticationURL());
        wv.requestFocus();
        btn.setClickable(true);
    }
}

File: app/src/main/java/jp/senchan/android/wasatter/task/TaskGetOAuthRequestUrl.kt
/**
 *
 */
package jp.senchan.android.wasatter.task

import android.os.AsyncTask
import android.view.View
import android.webkit.WebView
import android.widget.Button
import jp.senchan.android.wasatter.R
import jp.senchan.android.wasatter.activity.OAuthToken
import twitter4j.TwitterException

/**
 * @author takuji
 */
class TaskGetOAuthRequestUrl (private val target: OAuthToken) : AsyncTask<Unit, Unit, Unit>() {
    private var btn: Button? = null

    override fun onPreExecute() { // TODO 自動生成されたメソッド・スタブ
        btn = target.findViewById<View>(R.id.button_set_token) as Button
        btn!!.isClickable = false
    }

    override fun doInBackground(vararg params: Unit) { // TODO 自動生成されたメソッド・スタブ
        try {
            target.request = target.twitter!!.oAuthRequestToken
        } catch (e: TwitterException) { // TODO 自動生成された catch ブロック
            e.printStackTrace()
        }
    }

    override fun onPostExecute(result: Unit) { // TODO 自動生成されたメソッド・スタブ
        val wv = target.findViewById<View>(R.id.web) as WebView
        wv.loadUrl(target.request!!.authenticationURL)
        wv.requestFocus()
        btn!!.isClickable = true
    }

}